### PR TITLE
Improve app release documentation

### PR DESCRIPTION
**Pull request by GretaD**
1. git release remote has to be set up
2. document when to branch into stable branches
3. document when to bump the main version

### CODE REVIEW

File: processes/app_release.md [hunk 1]
```diff
@@ -33,6 +33,9 @@
 
 13. We should also push it to the release `git push release stablex.x --tags`
 
+    13.1 When you push the release for the first time, you might not have the release git repo, if you run the command and the error ` 'release' does not appear to be a git repository` popsup you need to run this command: `git remote add release git@github.com:nextcloud-releases/AppName.git`
```
Comment (ChristophWurst) [inline processes/app_release.md:36]: I would move that step into a prerequisite list

File: processes/app_release.md [hunk 2]
```diff
@@ -44,5 +47,6 @@
 15. Open the app's main repo and close the vx.y.z milestone (Issues -> Milestones in the Right Upper Corner)
 
 16. Create a milestone for the next patch version
-
-17. If the release was for patch version, forward-port the changelog to master/main (`git checkout stablex.x CHANGELOG.md`)
+17. If you are releasing a minor or major release, you need to branch off this minor or major release: `git checkout -b  stablex.x, push origin stablex.x`
```
Comment (ChristophWurst) [inline processes/app_release.md:50]: doesn't say where you branch off. this needs to be the target branch *after* the release branch has been merged and the local branch updated.

and split the two commands, now it looks like the checkout and push can be run at once.

File: processes/app_release.md [hunk 3]
```diff
@@ -44,5 +47,6 @@
...
+18. After branch off, you need to change the `info.xml` version to x.x.0-alpha.1, so add `alpha.1` to your branch off release version.
```
Comment (ChristophWurst) [inline processes/app_release.md:51]: clarification: this needs to be done in a PR against main

**Review by juliushaertl — COMMENTED** [review]
👍 for the remote part as that was what we encountered when doing the release

For the stable branches I guess it depends on if the groupware apps maintain dedicated stable branches for minor releases

File: processes/app_release.md [hunk 4]
```diff
@@ -33,20 +33,19 @@
...
 
-14. Then we go to the release's repository tags page, e.g. https://github.com/nextcloud-releases/mail/tags
+	17.1 On this draft we add the tag we created, add the change log and publish the release
 
-    14.1 On this draft we add the tag we created, add the change log and publish the release
-
-    14.2 Wait a few minutes for the automation to build the app
+    17.2 Wait a few minutes for the automation to build the app
     
-    16.3 The package will be atuomatically built and attached to the new release
```
Comment (miaulalala) [inline processes/app_release.md:42]: :see_no_evil:

**Review by miaulalala — APPROVED** [review]

File: processes/app_release.md (hunk 1 above)
Comment (ChristophWurst) [reply processes/app_release.md]: > GretaD marked this conversation as resolved. 

you only changed the indentation, right?

Comment (GretaD) [reply processes/app_release.md]: yes, and changed the number from 13.1 to 14.
so from 
1
  1.2
i did
1
2
did i get something wrong?

Comment (GretaD) [reply processes/app_release.md]: yes, and changed the number from 13.1 to 14.
so from 
1
  1.2
i did
1
2
did i get something wrong?

Comment (ChristophWurst) [reply processes/app_release.md]: https://docs.nextcloud.com/server/stable/admin_manual/installation/source_installation.html#prerequisites-for-manual-installation

It should be a separate list on the top

```

## Prerequisites
* git `release` origin, e.g. `git@github.com:nextcloud-release/mail.git`

## Process
1. ...
```

Comment (GretaD) [reply processes/app_release.md]: and today i learned something :)

File: processes/app_release.md [hunk 5]
```diff
@@ -33,16 +33,19 @@
 
 13. We should also push it to the release `git push release stablex.x --tags`
 
-14. Then we go to the release's repository tags page, e.g. https://github.com/nextcloud-releases/mail/tags
+14. When you push the release for the first time, you might not have the release git repo, if you run the command and the error ` 'release' does not appear to be a git repository` popsup you need to run this command: `git remote add release git@github.com:nextcloud-releases/AppName.git`
+15. Run point 13 again
+16. After releasing a minor or major version, you need to branch off this minor or major release: `git checkout -b  stablex.x` the do `push origin stablex.x`
```
Comment (st3iny) [inline processes/app_release.md:38]:
```suggestion
16. After releasing a minor or major version, you need to branch off this minor or major release: `git checkout -b  stablex.x` then do `git push origin stablex.x`
```

**Comment by st3iny** [comment]
> After branch off, you need to create a new separate PR to change the info.xml version to x.x.0-alpha.1, so add alpha.1 to your branch off release version.

Could someone elaborate on that step? Do I have to add the `alpha.1` suffix to stablex.x or main?

**Comment by ChristophWurst** [comment]
To main

File: processes/app_release.md [hunk 6]
```diff
@@ -33,16 +33,19 @@
...
+    17.3 The package will be automatically built and attached to the new release
 
-15. Open the app's main repo and close the vx.y.z milestone (Issues -> Milestones in the Right Upper Corner)
+19. Open the app's main repo and close the vx.y.z milestone (Issues -> Milestones in the Right Upper Corner)
 
-16. Create a milestone for the next patch version
-
-17. If the release was for patch version, forward-port the changelog to master/main (`git checkout stablex.x CHANGELOG.md`)
+20. Create a milestone for the next patch version
+21. After branch off, you need to create a new separate PR to change the `info.xml` version to x.x.0-alpha.1, so add `alpha.1` to your branch off release version.
```
Comment (st3iny) [inline processes/app_release.md:50]:
```suggestion
21. After branch off, you need to create a new separate PR against main to change the `info.xml` version to x.x.0-alpha.1, so add the suffix `-alpha.1` to your branch off release version.
```

**Comment by GretaD** [comment]
appreleaseproccess

**Review by st3iny — APPROVED** [review]

**Comment by st3iny** [comment]
Please squash all fixups and we are good to go.

**Comment by GretaD** [comment]
> The backport to main failed. Please do this backport manually.

nice try

**Pull request merged**